### PR TITLE
[sfmColorHarmonize] update selection method enum and command-line argument

### PR DESCRIPTION
**Pull request by mugulmd**
## Description

In this PR we update the `enum` system used for the `selectionMethod` argument of `sfmColorHarmonize` to use an `enum class` that works the same way as other `aliceVision` executables instead.

### CODE REVIEW

File: src/software/utils/sfmColorHarmonize/colorHarmonizeEngineGlobal.cpp [hunk 1]
```diff
@@ -427,7 +437,8 @@ bool ColorHarmonizationEngineGlobal::Process()
       }
     }
 
-    const std::string out_folder = (fs::path(_outputDirectory) / (vec_selectionMethod[ _selectionMethod ] + "_" + vec_harmonizeMethod[ harmonizeMethod ])).string();
+    const int selectionMethodIdx = static_cast<int>(_selectionMethod);
+    const std::string out_folder = (fs::path(_outputDirectory) / (vec_selectionMethod[ selectionMethodIdx ] + "_" + vec_harmonizeMethod[ harmonizeMethod ])).string();
```
Comment (fabiencastan) [inline src/software/utils/sfmColorHarmonize/colorHarmonizeEngineGlobal.cpp:441]: `vec_selectionMethod[ selectionMethodIdx ]` => enumToString

Comment (fabiencastan) [reply src/software/utils/sfmColorHarmonize/colorHarmonizeEngineGlobal.cpp:441]: remove all usages of vec_selectionMethod and vec_selectionMethod

Comment (mugulmd) [reply src/software/utils/sfmColorHarmonize/colorHarmonizeEngineGlobal.cpp]: Done

File: src/software/utils/main_sfmColorHarmonize.cpp [hunk 2]
```diff
@@ -57,10 +57,8 @@ int aliceVision_main( int argc, char **argv )
       "Path to folder(s) in which computed matches are stored.")
     ("referenceImage", po::value<int>(&imgRef)->required(),
       "Reference image id.")
-    ("selectionMethod", po::value<int>(&selectionMethod)->required(),
-      "- 0: FullFrame\n"
-      "- 1: Matched Points\n"
-      "- 2: VLD Segment");
+    ("selectionMethod", po::value<EHistogramSelectionMethod>(&selectionMethod)->required(), 
+      "Histogram selection method");
```
Comment (fabiencastan) [inline src/software/utils/main_sfmColorHarmonize.cpp:61]: Add `EHistogramSelectionMethod_description()` with the list of options, else we cannot know the list of possible values.

**Review by fabiencastan — CHANGES_REQUESTED** [review]
missing description of the option

File: src/software/utils/main_sfmColorHarmonize.cpp (hunk 2 above)
Comment (mugulmd) [reply src/software/utils/main_sfmColorHarmonize.cpp]: Done

**Review by fabiencastan — APPROVED** [review]

File: src/software/utils/sfmColorHarmonize/colorHarmonizeEngineGlobal.hpp [hunk 3]
```diff
@@ -16,13 +16,26 @@
...
-    eHistogramHarmonizeVLDSegment    = 2,
+    eHistogramHarmonizeFullFrame = 0,
+    eHistogramHarmonizeMatchedPoints,
+    eHistogramHarmonizeVLDSegment
 };
 
+inline std::string EHistogramSelectionMethod_description()
+{
+  return "Histogram selection method: \n"
+         "* full frame \n"
```
Comment (fabiencastan) [inline src/software/utils/sfmColorHarmonize/colorHarmonizeEngineGlobal.hpp:29]: Need the real valid values for that option, people cannot invent that it needs an underscore (like "full_frame").

**Review by fabiencastan — CHANGES_REQUESTED** [review]

File: src/software/utils/sfmColorHarmonize/colorHarmonizeEngineGlobal.hpp (hunk 3 above)
Comment (mugulmd) [reply src/software/utils/sfmColorHarmonize/colorHarmonizeEngineGlobal.hpp]: Fixed :)

**Pull request merged**